File: go-client-sdk/pkg/models/operations/postcoupons.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PostCouponsRequest struct {
	Request *shared.CouponInput `request:"mediaType=application/json"`
}

type PostCouponsResponse struct {
	ContentType string
	StatusCode  int
}

File: typescript-client-sdk/src/sdk/models/operations/postcoupons.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostCouponsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.CouponInput;
}


export class PostCouponsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
